python/spglib/spglib.py

```diff
@@ -97,6 +107,82 @@ class SpglibError:
 spglib_error = SpglibError()
 
 
+@dataclasses.dataclass(eq=True, frozen=True)
+class DictInterface(Mapping[str, "Any"], ABC):
+    """Base class for dataclass with dict interface.
+
+    .. versionadded:: 2.5.0
+    .. deprecated:: 2.5.0
+        Dict-like interface (``obj['field']``) are deprecated.
+        Please use attribute interface instead (``obj.field``)
+    """
+
+    def __getitem__(self, key: str) -> Any:
+        """Return the value of the key."""
+        warnings.warn(
+            f"dict interface ({self.__class__.__name__}['{key}']) is deprecated."
+            "Use attribute interface ({self.__class__.__name__}.{key}) instead",
+            DeprecationWarning,
+        )
+        return dataclasses.asdict(self)[key]
+
+    def __len__(self) -> int:
+        """Return the number of fields."""
+        return len(dataclasses.fields(self))
+
+    def __iter__(self) -> Iterator[str]:
+        """Return an iterator over the keys."""
+        return iter(dataclasses.asdict(self))
+
+
+@dataclasses.dataclass(eq=True, frozen=True)
+class SpaceGroupType(DictInterface):
+    """Space group type information."""
+
+    number: int
+    """International space group number"""
+    international_short: str
+    """International short symbol"""
+    international_full: str
+    """International full symbol"""
+    international: str
+    """International symbol"""
+    schoenflies: str
+    """Schoenflies symbol"""
+    hall_number: int
+    """Hall symbol serial number"""
+    hall_symbol: str
+    """Hall symbol"""
+    choice: str
+    """Centring, origin, basis vector setting"""
+    pointgroup_international: str
+    """International symbol of crystallographic point group"""
+    pointgroup_schoenflies: str
+    """Schoenflies symbol of crystallographic point group"""
+    arithmetic_crystal_class_number: int
+    """Arithmetic crystal class number"""
+    arithmetic_crystal_class_symbol: str
+    """Arithmetic crystal class symbol"""
+
+
+@dataclasses.dataclass(eq=True, frozen=True)
+class MagneticSpaceGroupType(DictInterface):
+    """Magnetic space group type information."""
+
+    uni_number: int
+    """Serial number of UNI (or BNS) symbols"""
+    litvin_number: int
+    """serial number in Litvin's [Magnetic group tables](https://www.iucr.org/publ/978-0-9553602-2-0)"""
```

Comment:
```suggestion
    """Serial number in Litvin's [Magnetic group tables](https://www.iucr.org/publ/978-0-9553602-2-0)"""
```
Small consistency change